### DISCUSSION TITLE
Add Manager Signals for Callbacks

### DISCUSSION THREAD

**Discussion by LennyPhoenix** [LennyPhoenix/GodotDiscordSDK#3] [Ideas]
Some manager's functions return results via callbacks, which are specified by passing a callback object, and a callback method name to the function. Another alternative solution could be to add signals to the manager classes that return these results. This would allow for the `yield` keyword to be used, reducing the amount of functions required, but would require a bit of work to get working.
We could potentially keep support for the old callback system, allowing use of both signals and pass-through callbacks.
What are people's thoughts on this?

**Comment by JezSonic** [comment]
I think adding signals to manager classes would be better or also, as you said `keep support for the old callback system`, could be a nice thing. In my performance signals are easier to code in GDScript, c# and Godot GDNative using C++ (I haven't tried to make signals in GDNative using C)

**Reply by LennyPhoenix** [reply]
Yeah, I think once I finish the other things on [the 1.0 board](https://github.com/LennyPhoenix/GodotDiscordSDK/projects/1) im going to add corresponding signals for each callback.

**Comment by LennyPhoenix** [comment]
Started work on this now 👌

**Reply by LennyPhoenix** [reply]
This is now fully implemented.
